src/FSize.h

```diff
@@ -78,67 +99,87 @@ class FSize {
      **/
     static const char * unit( const Unit unit_r ) {
       switch ( unit_r ) {
-      case T: return "TB";
-      case G: return "GB";
-      case M: return "MB";
-      case K: return "kB";
-      case B: break;
+      case Unit::Y: return "YiB";
+      case Unit::Z: return "ZiB";
+      case Unit::E: return "EiB";
+      case Unit::P: return "PiB";
+      case Unit::T: return "TiB";
+      case Unit::G: return "GiB";
+      case Unit::M: return "MiB";
+      case Unit::K: return "KiB";
+      case Unit::B: break;
       }
       return "B";
     }
 
   public:
 
     /**
-     * Construct from size in Byte.
+     * Construct from size in certain unit.
+     * E.g. <code>FSize( 1, FSize::Unit::K )<code> makes 1024 Byte.
      **/
-    FSize( const long long size_r = 0 )
-      : _size( size_r )
+    FSize( const boost::multiprecision::cpp_int &size_r = 0, const Unit unit_r = Unit::B)
+      : _size( boost::multiprecision::cpp_int(size_r) * factor( unit_r ) )
     {}
 
     /**
-     * Construct from size in certain unit.
-     * E.g. <code>FSize( 1, FSize::K )<code> makes 1024 Byte.
+     * Construct from size in Byte.
+     * @param size_r the initial value
      **/
-    FSize( const long long size_r, const Unit unit_r )
-      : _size( size_r * factor( unit_r ) )
+    FSize( double size_r )
+      : _size( size_r )
     {}
 
     /**
       Construct from string containing a number in given unit.
+      * @param sizeStr input string - must contain *only* numbers
+      * @param unit_r  optional unit, bytes by default
+      * @throws std::runtime_error if the string contains any non numeric characters,
+      *   even a trailing white space!
     */
-    FSize( const std::string &sizeStr, const Unit unit_r = B );
+    FSize( const std::string &sizeStr, const Unit unit_r = Unit::B );
 
     /**
-     * Conversion to long long
+     * Conversions to native data types - only explicit as it might overflow
+     * If the value is out of range then the behavior depends on the boost version
+     *   - 1.67 - the min/max values for the corresponding type are returned
+     *   - 1.66 - returns just the lower bits
```

Comment:
Huh, OK, and AFAICT Leap 15 has Boost 1.66.

Reply:
Yes, I found the problem when Travis built the package against TW which contains a newer boost (1.67 currently) than in Leap 15 (1.66) and an unit test failed.